### PR TITLE
Implemented Channels.list.joined api 

### PR DESCRIPTION
**Pull request by cloverhearts**
Rocket.chat api document.
- https://rocket.chat/docs/developer-guides/rest-api/channels/list-joined

- [x] api feature
- [x] document

Thank you :)

### CODE REVIEW

File: docs/channels.md [hunk 1]
```diff
@@ -132,6 +132,18 @@ session = rocket_server.login('username', 'password')
 channels = session.channels.list(query: {usernames: 'friend-username'})
 ```
 
+#### channels.list.joined
+
+Lists all of the channels the calling user has joined. 
+list is also used for searching/querying_
```
Comment (abrom) [inline docs/channels.md]: Please fix leading lower case and trailing `_`

File: docs/channels.md [hunk 2]
```diff
@@ -132,6 +132,18 @@ session = rocket_server.login('username', 'password')
...
+
+```ruby
+require 'rocketchat'
+
+rocket_server = RocketChat::Server.new('http://your.server.address/')
+session = rocket_server.login('username', 'password')
+channels = session.channels.list_joined(query: {usernames: 'friend-username'})
```
Comment (abrom) [inline docs/channels.md]: Please take a look at https://github.com/bbatsov/ruby-style-guide#spaces-braces

Specifically putting spaces inside hashes

File: lib/rocket_chat/messages/channel.rb [hunk 3]
```diff
@@ -40,6 +40,25 @@ def list(offset: nil, count: nil, sort: nil, fields: nil, query: nil)
...
+      # @param [Integer] count Query count/limit
+      # @param [Hash] sort Query field sort hash. eg `{ msgs: 1, name: -1 }`
+      # @param [Hash] fields Query fields to return. eg `{ name: 1, ro: 0 }`
+      # @param [Hash] query The query. `{ active: true, type: { '$in': ['name', 'general'] } }`
+      # @return [Room[]]
+      # @raise [HTTPError, StatusError]
+      #
+      def list_joined(offset: nil, count: nil, sort: nil, fields: nil, query: nil)
+        response = session.request_json(
+            '/api/v1/channels.list.joined',
```
Comment (abrom) [inline lib/rocket_chat/messages/channel.rb]: https://github.com/bbatsov/ruby-style-guide#spaces-indentation

**Comment by cloverhearts** [comment]
Thank you for  your review, i will check it :)

File: docs/channels.md (hunk 2 above)
Comment (cloverhearts) [reply docs/channels.md]: I have suggest answer for your opinion.
```
channels = session.channels.list_joined(query: { usernames: 'friend-username' })
```
what do you think?
actually, i have little bit confusing.
I mean, that's document code to same with [channels list api doc](https://github.com/abrom/rocketchat-ruby/blob/master/docs/channels.md#channelslist)

Comment (abrom) [reply docs/channels.md]: That looks right. Regarding the channels list api doc, that one needs to be fixed ;)

**Comment by abrom** [comment]
Reminder to also add some spec tests for this method. Take a look at https://github.com/abrom/rocketchat-ruby/blob/master/spec/messages/channel_spec.rb#L3 and https://github.com/abrom/rocketchat-ruby/blob/master/spec/messages/user_spec.rb#L239 for some inspiration

File: docs/channels.md (hunk 2 above)
Comment (cloverhearts) [reply docs/channels.md]: yeah~ Thanks, i will fix it, soon :)

**Comment by abrom** [comment]
Hi @cloverhearts, have you been able to take a look at the specs for these changes??

**Pull request closed without merging**